search-resources-api/src/test/java/no/unit/nva/search2/SwsResourceHandlerTest.java

```diff
@@ -0,0 +1,317 @@
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.collection.IsEmptyCollection.empty;
+import static org.hamcrest.core.Is.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class SwsResourceHandlerTest {
+
+    public static final String SAMPLE_PATH = "search";
+    public static final String SAMPLE_DOMAIN_NAME = "localhost";
+    public static final String SAMPLE_SEARCH_TERM = "searchTerm";
+    public static final String SAMPLE_OPENSEARCH_RESPONSE_WITH_AGGREGATION_JSON = "sample_opensearch_response.json";
+    public static final String ROUNDTRIP_RESPONSE_JSON = "roundtripResponse.json";
+    public static final String EMPTY_OPENSEARCH_RESPONSE_JSON = "empty_opensearch_response.json";
+    private SwsPagedResourceHandler handler;
+    private Context contextMock;
+    private ByteArrayOutputStream outputStream;
+    private SwsOpenSearchClient mockedSearchClient;
+
+    @BeforeEach
+    void setUp() {
+
+        mockedSearchClient = mock(SwsOpenSearchClient.class);
+        handler = new SwsPagedResourceHandler(new Environment(), mockedSearchClient);
+        contextMock = mock(Context.class);
+        outputStream = new ByteArrayOutputStream();
+    }
+
+    @Test
+    void shouldReturnSearchResultsWhenQueryIsSingleTerm() throws IOException {
+        prepareRestHighLevelClientOkResponse();
+
+        handler.handleRequest(getInputStream(), outputStream, contextMock);
+
+        var gatewayResponse = GatewayResponse.of(outputStream);
+        var actualBody = gatewayResponse.body();
+        var expected = getSearchResourcesResponseFromFile(ROUNDTRIP_RESPONSE_JSON);
+
+        assertNotNull(gatewayResponse.headers());
+        assertEquals(HTTP_OK, gatewayResponse.statusCode());
+        assertThat(actualBody.hits(), is(equalTo(expected.hits())));
+    }
+
+    @Test
+    void shouldReturnSortedSearchResultsWhenSendingContributorId() throws IOException {
+        prepareRestHighLevelClientOkResponse();
+        handler.handleRequest(getInputStreamWithContributorId(), outputStream, contextMock);
+
+        var gatewayResponse = GatewayResponse.of(outputStream);
+
+        assertNotNull(gatewayResponse.headers());
+        assertEquals(HTTP_OK, gatewayResponse.statusCode());
+    }
+
+    @Test
+    void shouldSearchResultsWhenSendingContributorIdAndBadResponseFromPreferencesApi()
+        throws IOException {
+        prepareRestHighLevelClientOkResponse();
+        handler.handleRequest(getInputStreamWithContributorId(), outputStream, contextMock);
+
+        var gatewayResponse = GatewayResponse.of(outputStream);
+
+        assertNotNull(gatewayResponse.headers());
+        assertEquals(HTTP_OK, gatewayResponse.statusCode());
```

Comment:
This is duplicate code from the test above

Reply:
removed all duplicates